test/libsolidity/syntaxTests/inheritance/override/public_var_no_override_but_function.sol

```diff
@@ -0,0 +1,8 @@
+contract A {
+	function foo() internal virtual pure returns(uint) { return 5; }
+}
+contract X is A {
+	uint public foo;
+}
+// ----
+// TypeError: (100-115): Base function must be external when overridden by public state variable.
```

Comment:
Still not too happy with this error message, but it might be fine. At least the error is on the state variable.

Reply:
Could change it to `Public state variable cannot override non-external function.`

Reply:
or `can only override external function`

Reply:
(`functions with external visibility`

Reply:
I'm not happy with it because at this point, we do not yet know whether we have the 'override' specifier given or not.

Reply:
If it is specified, then the message is correct.
If it's not, then when they fix the external function, the new message `override required here` will appear, so I guess also fine